Add statistics calculation for database records

NightMare489 opened the pull request:
Implemented the get_statistics() method to calculate various statistics from database records. This includes:
- Calculating the average age of records.
- Computing the distribution of levels (easy, normal, hard) as percentages.
- Tracking the frequency of important and non-important records.
- Tracking the frequency of open and closed statuses.
- Analyzing the most common words in the message fields.

NightMare489 committed:
Add statistics calculation for database records

Implemented the get_statistics() method to calculate various statistics from database records. This includes:
- Calculating the average age of records.
- Computing the distribution of levels (easy, normal, hard) as percentages.
- Tracking the frequency of important and non-important records.
- Tracking the frequency of open and closed statuses.
- Analyzing the most common words in the message fields.

DevAbdoTolba commented on back/app/models.py:
```diff
@@ -62,4 +66,75 @@ def from_dict(cls, data):
+            "level_distribution": level_distribution,
+            "important_frequency": important_frequency,
+            "status_frequency" : status_frequency,
+            "common_words": list(common_words.items())
```
Not all words are common_words. please select only the top %20 most common words in the whole database

also make sure to return a List of tuples 
```py
common_words_in_msg = [
    ("break",3),
    ("new", 5)
]
```
**not!!** 
`common_words_in_msg = [
    ["break",3],
    ["new", 5]
]`

DevAbdoTolba commented on back/app/models.py:
```diff
@@ -62,4 +66,75 @@ def from_dict(cls, data):
+
+        #calculating percentages
+        avg_age *= 100/records_len
+        avg_age = round(avg_age,1)
```
unattended behavior for age. calculate the average date from all the dates issued in the system.
2020-04-01
2020-04-02
2020-04-03

then the average days is 2 days is the distribution of work hours

DevAbdoTolba commented on back/app/models.py:
```diff
@@ -62,4 +66,75 @@ def from_dict(cls, data):
+        level_distribution={
+            "easy": 0,
+            "normal": 0,
+            "hard": 0
```
wrong formating. "easy" : [numberOfEasy, precentageOfEasy]
so it becomes : 
level_distribution = {
    "easy" : [4,20],
    "normal" : [6,30],
    "hard" : [10,50]
}

DevAbdoTolba reviewed (changes requested):
all good. requiring some changes. pleas make sure to follow up with the format : 

avg_age = 14.3
level_distribution = {
    "easy" : [4,20],
    "normal" : [6,30],
    "hard" : [10,50]
}

important_frequency = {
    True : 7,
    False : 13
}

status_frequency = {
    "open" : 16,
    "close" : 4
}

common_words_in_msg = [
    ("break",3),
    ("new", 5)
]

NightMare489 committed:
Modify statistics calculation function to get intended behavior fixed #25

- Made common_words only selects top 25% of the whole database.
-Fixed avg_age calculation.
-Corrected format for level_distribution to reflect frequency and percentage.

DevAbdoTolba reviewed (changes requested):
- [ ] common words "hello!" is condired as one word which is counted along side "hello", also "You" is not filtered as Y is capital. also filter `it`
- [ ] common words use the top 75% without the +1
- [ ] status calculated for "open" and "closed" but it is `close` please also update the serlizer
- [ ] round the number of avg_age to 0.00

NightMare489 committed:
Fixed issues not handled in previous commit

- Fixed issue with removing punctuation was not working properly.
- Added "it" to the filter of stop words
- Common words no longer use +1, so it can calculate correctly.
- "closed" was renamed to "close".
-avg_age now rounds to two decimal places.

DevAbdoTolba reviewed (approved):
LGTM 👍
